### PR TITLE
Internet shop

### PR DESCRIPTION
**Pull request by KudAndrii**

### CODE REVIEW

File: InternetShop/InternetShop/Cart.cs [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace InternetShop
+{
+    /// <summary>
+    /// Shopping cart, contains List of Iproduct (ShoppingCart).
+    /// </summary>
+    internal class Cart
```
Comment (DmytroZhuravlov14) [inline InternetShop/InternetShop/Cart.cs:12]: Cart contains ShoppingCart - looks wierd a bit. Expected for Cart to contain other types of carts?))

File: InternetShop/InternetShop/IProduct.cs [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace InternetShop
+{
+    /// <summary>
+    /// Contains price, name and description.
+    /// </summary>
+    internal interface IProduct
```
Comment (DmytroZhuravlov14) [inline InternetShop/InternetShop/IProduct.cs:12]: Not sure we already know aboout the interfaces. Anyway, interfaces should be in a separate folder, as enums, services and models

**Review by DmytroZhuravlov14 — CHANGES_REQUESTED** [review]

File: InternetShop/InternetShop/ShopCore/ShopService.cs [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+{
+    internal class ShopService
+    {
+        /// <summary>
+        /// Method adding products to incoming cart if input == name of some product of productRange.
+        /// </summary>
+        /// <param name="productRange">Incoming list of range of products.</param>
+        /// <param name="orderFormation">Example of class OrderFormation to get the list of products in cart.</param>
+        /// <param name="input">Incoming string to comparison.</param>
+        public void AddProduct(List<Product> productRange, ref OrderFormation orderFormation, string input)
```
Comment (DmytroZhuravlov14) [inline InternetShop/InternetShop/ShopCore/ShopService.cs:17]: For what reasons do we need ref here?

File: InternetShop/InternetShop/Program.cs [hunk 4]
```diff
@@ -0,0 +1,13 @@
+﻿using System;
+using System.Collections.Generic;
+
+namespace InternetShop
+{
+    internal class Program
+    {
+        private static void Main(string[] args)
+        {
+            new ShopInterface().ShopMenu(ProductRange.Instance.Products);
```
Comment (DmytroZhuravlov14) [inline InternetShop/InternetShop/Program.cs:10]: How do you think - would we need this new instance of shopInterface in future? IF so, create a separate variable above and use it further

**Review by DmytroZhuravlov14 — APPROVED** [review]

File: InternetShop/InternetShop/Program.cs (hunk 4 above)
Comment (KudAndrii) [reply InternetShop/InternetShop/Program.cs:10]: For now - no, but if situation changes - we can correct it very simple - write it to variable.
You mean it, don't you?

File: InternetShop/InternetShop/ShopCore/ShopService.cs (hunk 3 above)
Comment (KudAndrii) [reply InternetShop/InternetShop/ShopCore/ShopService.cs]: You are right, it is unnessessary

File: InternetShop/InternetShop/Program.cs (hunk 4 above)
Comment (DmytroZhuravlov14) [reply InternetShop/InternetShop/Program.cs:10]: Fine for now, just asking +_+

**Pull request merged**